lib/cloud/azure/sql_test.go

```diff
@@ -0,0 +1,139 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azure
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/sql/armsql"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSQLListAll(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	for _, tc := range []struct {
+		desc            string
+		client          armSQLServerClient
+		expectErr       require.ErrorAssertionFunc
+		expectedServers []string
+	}{
+		{
+			desc: "servers",
+			client: &ARMSQLServerMock{AllServers: []*armsql.Server{
+				makeSQLServer(t, "server1", "group1"),
+				makeSQLServer(t, "server2", "group2"),
+				makeSQLServer(t, "server3", "group1"),
+			}},
+			expectErr: require.NoError,
+			expectedServers: []string{
+				"server1.database.windows.net",
+				"server2.database.windows.net",
+				"server3.database.windows.net",
+			},
+		},
+		{
+			desc:            "empty list",
+			client:          &ARMSQLServerMock{AllServers: []*armsql.Server{}},
+			expectErr:       require.NoError,
+			expectedServers: []string{},
+		},
+		{
+			desc:            "auth error",
+			client:          &ARMSQLServerMock{NoAuth: true},
+			expectErr:       require.Error,
+			expectedServers: []string{},
+		},
+	} {
+		t.Run(tc.desc, func(t *testing.T) {
+			c := NewSQLClientByAPI(tc.client)
+
+			servers, err := c.ListAll(ctx)
+			tc.expectErr(t, err)
+			requireSQLServers(t, tc.expectedServers, servers)
+		})
+	}
+}
+
+func TestSQLListWithinGroup(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	for _, tc := range []struct {
+		desc            string
+		client          armSQLServerClient
+		expectErr       require.ErrorAssertionFunc
+		expectedServers []string
+	}{
+		{
+			desc: "servers",
+			client: &ARMSQLServerMock{ResourceGroupServers: []*armsql.Server{
+				makeSQLServer(t, "server1", "group1"),
+				makeSQLServer(t, "server2", "group1"),
+				makeSQLServer(t, "server3", "group1"),
+			}},
+			expectErr: require.NoError,
+			expectedServers: []string{
+				"server1.database.windows.net",
+				"server2.database.windows.net",
+				"server3.database.windows.net",
+			},
+		},
+		{
+			desc:            "empty list",
+			client:          &ARMSQLServerMock{ResourceGroupServers: []*armsql.Server{}},
+			expectErr:       require.NoError,
+			expectedServers: []string{},
+		},
+		{
+			desc:            "auth error",
+			client:          &ARMSQLServerMock{NoAuth: true},
+			expectErr:       require.Error,
+			expectedServers: []string{},
+		},
+	} {
+		t.Run(tc.desc, func(t *testing.T) {
+			c := NewSQLClientByAPI(tc.client)
+
+			servers, err := c.ListWithinGroup(ctx, "group1")
+			tc.expectErr(t, err)
+			requireSQLServers(t, tc.expectedServers, servers)
+		})
+	}
+}
+
+func requireSQLServers(t *testing.T, expected []string, actual []*armsql.Server) {
+	t.Helper()
+
+	serverNames := []string{}
```

Comment:
nit: 
```suggestion
	var serverNames []string{}
```
https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices